blockmanager.go

```diff
@@ -2529,8 +2530,27 @@ func (b *blockManager) handleHeadersMsg(hmsg *headersMsg) {
 			})
 			totalWork := big.NewInt(0)
 			for j, reorgHeader := range msg.Headers[i:] {
-				err = b.checkHeaderSanity(reorgHeader,
-					maxTimestamp, true)
+				// We have to get the parent's height and
+				// header to be able to contextually validate
+				// this header.
+				prevNodeHeight := backHeight + uint32(j)
+
+				var prevNodeHeader *wire.BlockHeader
+				if i+j == 0 {
```

Comment:
🤔

Reply:
Aren't these both positive integers? or this is simply detecting the first iteration?

Reply:
Yes, this just means "the very first iteration".